PythonVisualizations/OrderedArray.py

```diff
@@ -317,16 +335,19 @@ def find(self, val):
                 deltaXMid = ((lo- hi) //2) -1
                 self.moveItemsBy(indexMid, (self.CELL_SIZE* deltaXMid, 0))
         
-        self.stopAnimations()
         self.window.update()
+        self.stopAnimations()
         self.cleanUp(callEnviron)
         return lo                         #val not found 
     
             
     def remove(self, val):
-        callEnviron = self.createCallEnvironment()
-        self.startAnimations()
+        callEnviron = self.createCallEnvironment()         
+    
         index = self.find(val)
+        self.startAnimations()
+        self.cleanUp(callEnviron)
```

Comment:
I think the previous initialization is better.  In other words, start with:
```
    def remove(self, val):
        callEnviron = self.createCallEnvironment()         
        self.startAnimations()
        index = self.find(val)
```
**Do not** keep the call to `self.cleanUp()` on line 349.  That's saying that the animation for this call has ended and it is time to clean up the temporary variables.  There is another call to  `self.cleanUp()` on line 377 that is in the right place.